mitodl/micromasters
# Added program grade to ES index to support filtering by grade

#### What are the relevant tickets?

Closes #682 

#### What's this PR do?

Adds 'grade_average' to the ES index (calculated at index time). With this, we can work on #731 (filtering users by program grade)

#### Where should the reviewer start?

Probably dashboard/serializers.py since that's where the program grade is calculated. Probably search/api.py after that

#### How should this be manually tested?

If you'd like, you can reindex and check the actual ES results using this query (Postman is very useful for this). This will filter by average grade >= 75 and build a histogram in 10 pt intervals:

```
{
    "query": {
        "nested": {
            "path": "program",
            "query": {
                "bool": {
                    "must": [
                        { "range": { "program.grade_average": { "gte": 75 } } }
                    ]
                }
            }
        }
    },
    "aggs": {
        "overall_grade": {
            "nested" : {
                "path" : "program"
            },
            "aggs": {
                "grade_hist" : {
                    "histogram" : {
                        "field" : "program.grade_average",
                        "interval" : 10
                    }
                }
            }
        }
    }
}
```
